### PR TITLE
ppc64le port.

### PR DESCRIPTION
**Pull request by asowani**
This file contains changes required to build and use it on ppc64le platform.

### CODE REVIEW

File: Sources/Basic/TerminalController.swift [hunk 1]
```diff
@@ -85,9 +85,11 @@ public final class TerminalController {
 
         // Try determining using ioctl.
         var ws = winsize()
+#if !arch(powerpc64le)
```
Comment (aciidb0mb3r) [inline Sources/Basic/TerminalController.swift]: Can you add a comment that this doesn't work on ppc. We also also include the above `ws` variable in this #if statement.

Comment (asowani) [reply Sources/Basic/TerminalController.swift]: @aciidb0mb3r I understand that TIOCGWINSZ is supported on ppc64le and I guess "sys/ioctl.h" should be included to get that definition. I would like to give it a try to see if that actually works on ppc64le. However, how should this file be included in the swift code? I see "import" statements at the beginning, so is "import sys/ioctl.h" correct swift statement? Sorry, I am not a swift developer, hence don't know much about it. Could you please help me with swift code?

Comment (aciidb0mb3r) [reply Sources/Basic/TerminalController.swift]: Swift uses Clang modulemaps for importing C headers. For system headers, there is a modulemap shipped with macOS and a modulemap in the Swift repository for Linux. I am not really sure if Swift has a modulemap for ppc. /cc @jrose-apple

Comment (aciidb0mb3r) [reply Sources/Basic/TerminalController.swift]: Also, the official Swift language book is an excellent resource for learning the language! The book is freely available as a digital copy. Here is the link: https://itunes.apple.com/us/book/the-swift-programming-language-swift-4-0-3/id881256329?mt=11

Comment (jrose-apple) [reply Sources/Basic/TerminalController.swift]: Module maps are by OS, not by architecture. I assume the ppc64le we're trying to bring up is a Linux, right?

Comment (asowani) [reply Sources/Basic/TerminalController.swift]: Yes, it is Linux.

**Comment by asowani** [comment]
While I figure out how to add ioctl.h file to swift code, is it acceptable if I move the "ws" inside ppc64le condition and commit the code?

**Comment by asowani** [comment]
Until I complete my work on ioctl.h include, I have moved "ws" inside ppc64le condition as suggested by @aciidb0mb3r

File: Sources/Basic/TerminalController.swift (hunk 1 above)
Comment (jrose-apple) [reply Sources/Basic/TerminalController.swift]: It looks like we've manually mapped [a number of the TIOC* constants](https://github.com/apple/swift/blob/master/stdlib/public/Platform/TiocConstants.swift) over in the "Glibc" module (already imported in this file), probably because they are macros that don't import well to Swift. (The rest of ioctl is probably already present.) But right now, the constants are limited to architectures that we know have them. Maybe Linux should be added to that list, at least for some of them?

This would involve figuring out which ones do not import on Linux, and whether the values listed are correct.

**Comment by aciidb0mb3r** [comment]
@asowani Does this patch make SwiftPM ppc compatible or is it an incremental fix? (It is fine either way, I am just curious)

File: Sources/Build/Triple.swift [hunk 2]
```diff
@@ -105,12 +106,15 @@ public struct Triple {
 
     public static let macOS = try! Triple("x86_64-apple-macosx10.10")
     public static let linux = try! Triple("x86_64-unknown-linux")
+    public static let ppc64leLnx = try! Triple("powerpc64le-unknown-linux")
```
Comment (xwu) [inline Sources/Build/Triple.swift]: Drive-by nit: could you spell this `ppc64leLinux`?

Comment (asowani) [reply Sources/Build/Triple.swift]: @xwu I have committed necessary changes today. Thanks!

**Comment by asowani** [comment]
@aciidb0mb3r These changes make SwiftPM compatible with ppc64le.

**Comment by aciidb0mb3r** [comment]
Sounds good. Thanks!

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Comment by asowani** [comment]
Now that the changes have been approved, is there anything further I need to do to get them merged? Please let me know. Thanks!

**Comment by aciidb0mb3r** [comment]
This looks good, thanks!
